vividus-plugin-rest-api/src/test/java/org/vividus/bdd/steps/api/HttpRequestStepsTests.java

```diff
@@ -149,6 +157,35 @@ void testSetCustomRequestConfigNoField()
         assertEquals("nonExistentField", exception.getMessage());
     }
 
+    @Test
+    void testWaitForResponseCode()
+    {
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        when(httpTestContext.getResponse()).thenReturn(new HttpResponse());
+        httpRequestSteps.waitForResponseCode(RESPONSE_CODE, Duration.ofSeconds(10), RETRY_TIMES, stepsToExecute);
+        verify(stepsToExecute, times(RETRY_TIMES)).execute(Optional.empty());
+    }
+
+    @Test
+    void testWaitForResponseCodeWhenResponseIsNull()
+    {
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        when(httpTestContext.getResponse()).thenReturn(null);
+        httpRequestSteps.waitForResponseCode(RESPONSE_CODE, Duration.ofSeconds(10), RETRY_TIMES, stepsToExecute);
+        verify(stepsToExecute, times(RETRY_TIMES)).execute(Optional.empty());
+    }
+
+    @Test
+    void testWaitForResponseCodeWhenResponseCodeIsEqualToExpected()
+    {
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        HttpResponse httpResponse = new HttpResponse();
+        httpResponse.setStatusCode(200);
+        when(httpTestContext.getResponse()).thenReturn(httpResponse);
```

Comment:
what about emulation of real behavior:
- 1 attempt - return http response with any code, but not 200
- 2 attempt - return http response with code 200 
?